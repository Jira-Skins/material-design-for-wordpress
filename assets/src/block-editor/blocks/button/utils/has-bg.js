export default style => style === 'unelevated' || style === 'raised';
