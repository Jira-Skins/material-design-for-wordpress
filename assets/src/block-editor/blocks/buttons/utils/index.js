import { MDCRipple } from '@material/ripple';

export const initButtons = () => {
	const buttons = document.querySelectorAll( '.mdc-button' );
	const iconButtons = document.querySelectorAll( '.mdc-icon-button' );

	buttons.forEach( button => new MDCRipple( button ) );
	iconButtons.forEach( button => ( new MDCRipple( button ).unbounded = true ) );
};
