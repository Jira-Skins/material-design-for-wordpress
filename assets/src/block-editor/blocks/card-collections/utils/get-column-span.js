/**
 * Get column span for grid.
 *
 * @param {string} style - Grid style.
 * @param {number} columns - Number of columns.
 * @return {number} Column span.
 */
export default ( style, columns ) => {
	let columnSpan = 12;

	if ( style === 'grid' ) {
		/*
		 * This works well for the design if we have a maximum of 4 columns. It would not work
		 * so well for 5 and 7 columns for example. Something to keep in mind if the max number of columns
		 * increase above 4.
		 */
		columnSpan = Math.floor( 12 / columns );
	}

	return columnSpan;
};
