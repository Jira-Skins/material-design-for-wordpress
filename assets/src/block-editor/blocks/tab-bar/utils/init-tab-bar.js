import { MDCTabBar } from '@material/tab-bar';

export const initTabBar = () => {
	const tabBars = document.querySelectorAll( '.mdc-tab-bar-container' );

	tabBars.forEach( tabBarContainer => {
		const tabBar = new MDCTabBar(
			tabBarContainer.querySelector( '.mdc-tab-bar' )
		);

		const contentElements = tabBarContainer.querySelectorAll(
			'.mdc-tab-content'
		);

		tabBar.listen( 'MDCTabBar:activated', event => {
			tabBarContainer
				.querySelector( '.mdc-tab-content--active' )
				.classList.remove( 'mdc-tab-content--active' );

			contentElements[ event.detail.index ].classList.add(
				'mdc-tab-content--active'
			);
		} );
	} );
};
