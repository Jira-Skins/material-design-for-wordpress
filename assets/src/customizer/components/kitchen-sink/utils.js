export const inlineFontFamily = font => `'${ font }', sans-serif`;

export const materialIconClass = style =>
	'material-icons' + ( style === 'filled' ? '' : '-' + style );
