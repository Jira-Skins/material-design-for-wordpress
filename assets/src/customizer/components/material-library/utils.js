export const inlineFontFamily = font => `'${ font }', sans-serif`;

export const materialIconClass = style =>
	'material-icons' + ( style === 'filled' ? '' : '-' + style );

export const materialIconFontName = style =>
	"'Material Icons" +
	( style === 'filled' ? "'" : ' ' + style.replace( '-', ' ' ) + "'" );
