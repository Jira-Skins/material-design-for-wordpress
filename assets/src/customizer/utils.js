/**
 * Sanitize control id.
 *
 * @param {string} id
 */
export const sanitizeControlId = id =>
	id.replace( /\]$/, '' ).replace( /\[|\]/g, '-' );
