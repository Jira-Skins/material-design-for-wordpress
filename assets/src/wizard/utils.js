/* globals mtbWizard, mtbOnboarding, fetch */
/**
 * Functions that don't change the state of the world
 */

/**
 * Install theme if necessary
 * Activate theme
 *
 */
export const handleThemeActivation = () => {
	const action = mtbWizard.themeStatus;
	if ( 'ok' === action ) {
		return;
	}

	const parameters = {
		method: 'POST',
		headers: { 'X-WP-Nonce': mtbOnboarding.nonce },
	};

	return new Promise( ( resolve, reject ) => {
		fetch( `${ mtbOnboarding.restUrl }${ action }-theme`, parameters )
			.then( response => response.json() )
			.then( data => {
				if ( 'install' === action ) {
					fetch( `${ mtbOnboarding.restUrl }activate-theme`, parameters )
						.then( response => response.json() )
						.then( resolve )
						.catch( error => reject( error ) );
				} else {
					if ( data.code ) {
						reject( data );
					}

					resolve( data );
				}
			} )
			.catch( error => reject( error ) );
	} );
};

/**
 * Import demo content
 *
 */
export const handleDemoImporter = () => {
	const parameters = {
		method: 'POST',
		headers: { 'X-WP-Nonce': mtbWizard.nonce },
	};

	return new Promise( ( resolve, reject ) => {
		fetch( `${ mtbWizard.restUrl }install-content`, parameters )
			.then( response => response.json() )
			.then( response => {
				if ( response.code ) {
					reject( response );
				} else {
					resolve( response );
				}
			} )
			.catch( error => reject( error ) );
	} );
};

/**
 * Redirect to settings location
 *
 * @param {*} data Request json response
 */
export const redirectToSettings = data => {
	if ( data && 'success' === data.status ) {
		return window.location.replace( mtbWizard.settingsUrl );
	}
};
