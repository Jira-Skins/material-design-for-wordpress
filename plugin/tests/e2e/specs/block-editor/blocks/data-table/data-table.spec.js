/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import {
	clickButton,
	createNewPost,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import { insertBlockByKeyword, selectBlockByName } from '../../../../utils';

const createButtonLabel = 'Create Table';
const buttonSelector = '.blocks-table__placeholder-form button';

describe( 'blocks: material/data-table core/table style', () => {
	beforeEach( async () => {
		await createNewPost( {} );
		await insertBlockByKeyword( 'Table' );
		await page.waitForSelector( buttonSelector );
		// Create the table.
		await clickButton( createButtonLabel );
	} );

	it( 'should be inserted', async () => {
		// Check if block was inserted
		expect( await page.$( '[data-type="core/table"]' ) ).not.toBeNull();
	} );

	it( 'should have a "Styles" selection with Material style added', async () => {
		await selectBlockByName( 'core/table' );
		const panels = await page.$$( '.components-panel__body-toggle' );

		if (
			'true' !==
			( await page.evaluate(
				el => el.getAttribute( 'aria-expanded' ),
				panels[ 0 ]
			) )
		) {
			await panels[ 0 ].click();
		}

		const styles = await page.$$(
			'.block-editor-block-styles__item-label'
		);
		expect( styles ).toHaveLength( 3 );
	} );

	it( 'should update table cell content', async () => {
		await page.waitForSelector( 'td' );
		await page.click( 'td' );
		await page.keyboard.type( 'Column 1' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
