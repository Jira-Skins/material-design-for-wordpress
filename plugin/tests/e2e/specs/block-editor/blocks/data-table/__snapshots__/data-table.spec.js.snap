// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: material/data-table core/table style should update table cell content 1`] = `
"<!-- wp:table {\\"className\\":\\"is-style-material\\"} -->
<div class=\\"wp-block-table is-style-material\\"><div class=\\"mdc-data-table\\"><table class=\\"mdc-data-table__table\\"><tbody class=\\"mdc-data-table__content\\"><tr class=\\"mdc-data-table__row\\"><td class=\\"mdc-data-table__cell\\">Column 1</td><td class=\\"mdc-data-table__cell\\"></td></tr><tr class=\\"mdc-data-table__row\\"><td class=\\"mdc-data-table__cell\\"></td><td class=\\"mdc-data-table__cell\\"></td></tr></tbody></table></div></div>
<!-- /wp:table -->"
`;
